test/engines/wintermute/path_utils.h

```diff
@@ -0,0 +1,229 @@
+#include <cxxtest/TestSuite.h>
+#include "engines/wintermute/utils/path_util.h"
+/**
+ * Test suite for the functions in engines/wintermute/utils/path_util.h
+ *
+ * NOTE: This is not a prescription;
+ *       this was not written by the original engine author;
+ *       this was not written by the engine porter.
+ *
+ * It might, however, help to spot variations in behavior that are introduced by modifications
+ */
+
+class PathUtilTestSuite : public CxxTest::TestSuite {
+	public:
+	const Common::String unixPath;
+	const Common::String unixCapPath;
+	const Common::String windowsPath;
+	const Common::String windowsCapPath;
+	const Common::String emptyString;
+	const Common::String dualExtPath;
+	const Common::String manyExtPath;
+	const Common::String mixedSlashesPath1;
+	const Common::String mixedSlashesPath2;
+	const Common::String unixRelativePath;
+	const Common::String windowsRelativePath;
+	PathUtilTestSuite () :
+		unixPath("/some/file.ext"),
+		unixCapPath("/SOME/FILE.EXT"),
+		windowsPath("C:\\some\\file.ext"),
+		windowsCapPath("C:\\SOME\\FILE.EXT"),
+		emptyString(""),
+		dualExtPath("/some/file.tar.gz"),
+		manyExtPath("/some/file.tar.bz2.gz.zip"),
+		mixedSlashesPath1("C:\\this/IS_REALLY\\weird.exe"),
+		mixedSlashesPath2("/pretty\\weird/indeed.txt"),
+		unixRelativePath("some/file.ext"),
+		windowsRelativePath("some\\file.ext")
+	{}
+	void test_getdirectoryname() {
+		TS_ASSERT_EQUALS(
+				Wintermute::PathUtil::getDirectoryName(unixPath),
+				Common::String("/some/")
+				);
+		TS_ASSERT_EQUALS(
+				Wintermute::PathUtil::getDirectoryName(unixCapPath),
+				Common::String("/SOME/")
+				);
+		TS_ASSERT_EQUALS(
+				Wintermute::PathUtil::getDirectoryName(windowsPath),
+				Common::String("C:\\some\\")
+				);
+		TS_ASSERT_EQUALS(
+				Wintermute::PathUtil::getDirectoryName(windowsCapPath),
+				Common::String("C:\\SOME\\")
+				);
+
+		/*
+		 * Failing assert in Common::Array
+		 */
+		// TS_ASSERT_EQUALS(
```

Comment:
I un-comment it in 76b49c8, after I've fixed the failing behaviour.

I can remove it altogether, I see how it can raise eyebrows but in this particukar circumstances it seemed clearer this way to me.